client/src/pages/Search.js

```diff
@@ -32,14 +33,30 @@ function Search() {
     const fetchCompanies = async (company) => {
         try {
             const encodedQuery = encodeURIComponent(company)
-            const data = await fetchWithAuth({
+            const profileData = await fetchWithAuth({
                 url: `${HOST}/api/pipeline/search/company/${encodedQuery}`,
                 method: 'GET',
             })
 
+            const companyData = await fetchWithAuth({
+                url: `${HOST}/api/company/get/${encodedQuery}`,
+                method: 'GET',
+            })
+
             // Assuming fetchWithAuth throws for non-OK responses, including 404
             setErrorMessage('') // If the fetch was successful, there's no 404, so we assume some results were found
-            setProfiles([...data]) // Update the profiles state with the fetched data
+            setProfiles([...profileData]) // Update the profiles state with the fetched data
+
+            const companyRating = Math.floor(
+                companyData.rating / companyData.ratedEmployees.length / 20
+            )
+            setCompany({
+                id: companyData.name,
+                name: companyData.displayName,
+                logo: companyData.logo,
+                info: companyData.description,
+                rating: companyRating,
+            }) // fetch company data to display in card
```

Comment:
You should handle a case where there is no ratedEmployees because you're passing in 'NaN' ratings to the company card by dividing by zero which is causing it display blank rating pictures for companies that have not been rated

Reply:
You got it G.O.A.T.